dart/tool/java/generate_java.dart

```diff
@@ -895,8 +897,13 @@ class TypeRef {
       if (isArray) {
         print('skipped accessor body for $propertyName');
       } else {
-        writer.addLine(
-            'return $name.valueOf(json.get("$propertyName").getAsString());');
+        writer
+            .addLine('String name = json.get("$propertyName").getAsString();');
+        writer.addLine('try {');
+        writer.addLine('  return $name.valueOf(name);');
+        writer.addLine('} catch (IllegalArgumentException e) {');
+        writer.addLine('  return $name.Unknown;');
```

Comment:
I thought about that, but wanted clients to explicitly handle (or rather think about) the case where an older client is communicating with a newer VM.